File: retort-annotations/src/main/lombok/co/kaioru/retort/annotation/adapter/DescriptionAnnotationProcessor.java
package co.kaioru.retort.annotation.adapter;

import co.kaioru.retort.ICommand;
import co.kaioru.retort.ICommandContext;
import co.kaioru.retort.annotation.AbstractAnnotationProcessor;
import co.kaioru.retort.annotation.type.Description;

import java.lang.reflect.Method;

public class DescriptionAnnotationProcessor<I extends ICommandContext, O> extends AbstractAnnotationProcessor<I, O> {

    @Override
    public void process(ICommand command, Object object, Method method) {
        if (method.isAnnotationPresent(Description.class))
            command.setDescription(method.getAnnotation(Description.class).value());
    }

}

File: retort-annotations/src/main/kotlin/co/kaioru/retort/annotation/impl/adapter/DescriptionAnnotationProcessor.kt
package co.kaioru.retort.annotation.impl.adapter

import co.kaioru.retort.ICommand
import co.kaioru.retort.annotation.Description
import co.kaioru.retort.annotation.impl.AnnotationProcessor
import java.lang.reflect.Method

class DescriptionAnnotationProcessor<I, O> : AnnotationProcessor<I, O>() {
    override fun process(command: ICommand<I, O>, any: Any, method: Method) {
        if (method.isAnnotationPresent(Description::class.java))
            command.description = method.getAnnotation(Description::class.java).value
    }
}